src/trellis/core/message_handler.py

```
@@ -252,6 +260,33 @@ async def _invoke_callback(self, callback_id: str, args: list[tp.Any]) -> None:
+            await asyncio.sleep(self.batch_delay)
+
+            # Check if there are dirty nodes to render
+            if not self.tree.has_dirty_nodes():
+                continue
+
+            try:
+                patches = self.tree.render_and_diff()
+                if patches:
+                    await self.send_message(PatchMessage(patches=patches))
+            except Exception as e:
+                logger.exception(f"Error in render loop: {e}")
```

Ruff (0.14.10)

Redundant exception object included in `logging.exception` call

(TRY401)